src/Pate/Monad.hs

```diff
@@ -785,8 +862,34 @@ throwHere err = withValid $ do
 instance MF.MonadFail (EquivM_ sym arch) where
   fail msg = throwHere $ PEE.EquivCheckFailure $ "Fail: " ++ msg
 
-manifestError :: MonadError e m => m a -> m (Either e a)
-manifestError act = catchError (Right <$> act) (pure . Left)
+manifestError :: EquivM_ sym arch a -> EquivM sym arch (Either (PEE.EquivalenceError arch) a)
+manifestError act = do
+  catchError (Right <$> act) (pure . Left) >>= \case
+    r@(Left er) -> CMR.asks envFailureMode >>= \case
+      ThrowOnAnyFailure -> throwError er
+      ContinueAfterRecoverableFailures -> case PEE.isRecoverable (PEE.errEquivError er) of
+        True -> return r
+        False -> throwError er
+      ContinueAfterFailure -> return r
+    r -> return r
+
+internalizeError :: Either (PEE.EquivalenceError arch) a -> EquivM_ sym arch a
+internalizeError merr = case merr of
+  Left err -> throwError err
+  Right a -> return a
+
+-- | Run an IO operation, internalizing any exceptions raised
```

Comment:
What does "internalize" mean here?

Reply:
It's essentially the inverse of `manifestError` - raise an error if it exists and otherwise return. I'm happy to change it to either `unManifestError` or `raiseIfError`, whichever makes more sense.

Reply:
I think those are all still confusing (except maybe `raiseIfError`). The slightly non-obvious thing that I think is worth capturing is that this turns Haskell IO exceptions into (slightly cleaner?) `ExceptT` exceptions in `EquivM_`.

I suppose that is the definition of "internalize"

Reply:
I've removed it in favor of using the more standard 'finalize' function from Monad.Catch, which I think is more clear.